prereise/gather/griddata/hifld/data_process/transmission.py

```diff
@@ -303,8 +462,26 @@ def create_buses(lines):
     return buses
 
 
-def build_transmission():
-    """Main user-facing entry point."""
+def build_transmission(method="sub2line", kwargs={"rounding": 3}):
+    """Build transmission network
+
+    :param str method: method used to build network. Default method is *sub2line*
+        where all substations are considered and lines are filtered accordingly. Other
+        method is *line2sub* where all lines are considered and a list of substations
+        is compiled.
+    :param dict kwargs: keyword arguments to be passed to functions.
```

Comment:
Can we be more specific here, that `kwargs` is only used when `method == "line2sub"`?

Reply:
Sure. You don't expect any other keyword arguments for other function(s), e.g. (voltage, topology, etc.) in the future?

Reply:
If not, perhaps, I should only have a `rounding` optional argument with `3` as default value instead of unpacking a dictionary

Reply:
If we have multiple functions' optional keyword arguments contained in one big user-facing dict, then I think we'll need to change the call signature of the lower-level functions, right? So `map_lines_to_substations_using_coords(substations, lines, rounding=3)` would become `map_lines_to_substations_using_coords(substations, lines, **kwargs)`, and then internal to the function we'd have a dict of defaults including `rounding=3` that can supplement if the user-suppled `kwargs` doesn't specify, and then within the function we'd look up `kwargs["rounding"]`. Otherwise, if `map_lines_to_substations_using_coords` gets keyword arguments for other functions (of if they get `rounding`), it'll throw an error.

Either pattern should work I think. If we want to design `build_transmission` using the one-big-dict way, then let's future-proof the first lower-level function that ingests that dict, and illustrate a pattern for other lower-level functions. Otherwise, let's rename `kwargs` to something like `line2sub_kwargs`, or just pass `line2sub_rounding`.

Reply:
I went with the user-facing dictionary